sync/class.jetpack-sync-module-posts.php

```diff
@@ -316,76 +286,84 @@ function filter_post_content_and_add_links( $post_object ) {
 	}
 
 	public function save_published( $new_status, $old_status, $post ) {
-		if ( 'publish' === $new_status && 'publish' !== $old_status ) {
-			$this->just_published[ $post->ID ] = true;
-		}
 
-		$this->previous_status[ $post->ID ] = $old_status;
+
+		if ( ! isset( $this->sync_items[ $post->ID ] ) ) {
+			$this->sync_items[ $post->ID ] = new Jetpack_Sync_Item( 'save_post' );
+		}
+		$sync_item = $this->sync_items[ $post->ID ];
+		$just_published = 'publish' === $new_status && 'publish' !== $old_status;
+		$sync_item->add_meta( 'just_published', $just_published );
+		$sync_item->add_meta( 'previous_status', $old_status );
 	}
 
 	public function wp_insert_post( $post_ID, $post = null, $update = null ) {
 		if ( ! is_numeric( $post_ID ) || is_null( $post ) ) {
 			return;
 		}
 
+		if ( wp_is_post_revision( $post ) && $this->is_saving_post( $post->post_parent ) ) {
```

Comment:
Consider replacing this block of code with an alias function and a method to do the work, just to make things more orderly and compartmentalize them:
```
if ( $this->is_revision( $post ) {
  $this->process_revision( $post );
  return;
}
```